Codeinwp/woocommerce-product-addon
# Deselect Support for the ImageSelect Field

### Summary

Before this PR; de-select the selected **Image Select** field was possible only when "multiple selections" setting was activated, otherwise de-select was not possible.

**With this PR; de-select capability is added if the "non-multiple mode is enabled and the required mode is not enabled (if two condition met)".**

**Technically:**
* Behind the scene; Image Select was using Radio Input (for the nonmultiple case) and Checkbox field (for the multiple case). With this PR; I've removed the Radio Input and leveraged from the Checkbox Field. With this way; leveraged the native deselect capability of HTML Checkbox Element.
* Some attributes of radio input were moved into the checkbox element (only if multiple modes are off). That attributes (data-builder and custom attrs) probably is used for compatibility between PPOM Product Builder Addon. more: https://github.com/Codeinwp/ppom-pro/issues/105#issue-1465262327

**Expected Behavior:**
* If "multiple selections" setting is disabled and if the Image Select field is set as required: do not allow de-select and allow only max 1 selection.
* **If "multiple selections" setting is disabled and if the Image Select field is set as not-required: with this PR; de-select is possible.**
* If "multiple selections" setting is enabled and if the Image Select field is set as required; no change is expected. (multiple selection should be allowed, and de-select should be available)
* If "multiple selections" setting is enabled and if the Image Select field is set as not required; no change is expected. (multiple selections should be allowed, and de-select should be available)

### Will affect visual aspect of the product

No

### Screenshots


### Test instructions

Cases:
* No regression on the image select field (multiple selections mode is on/off, and required mode is on/off combinations)
* For the non-multiple select mode; deselect capability should be available only if the field is not required.
* No regression on Image Select Field (when multiple selection mode is off) overall (because a critical change has been made behind the scenes for non-multiple selections) on frontend, on order (order data is saved well)
* ImageSelect - PPOM Product Builder Addon compatibility preserved. (Note: the PPOM Product Builder Addon currently is not part of the PPOM Pro), there is an issue to follow up it: https://github.com/Codeinwp/ppom-pro/issues/105 - For this PR, we can skip that case only.

Screen GIF: https://vertis.d.pr/i/juWPph


Closes https://github.com/Codeinwp/ppom-pro/issues/48.
